### PR TITLE
feat(t32): initial support

### PR DESCRIPTION
**Pull request by xasc**
TRACE32 is a debugger for embedded software development that comes with a scripting language for automation. Here is a request to integrate the existing parser.

### CODE REVIEW

File: queries/t32/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  (time)
+] @number
+
+(float) @float
+
+(string) @string
+
+(path) @string.special
+
+(symbol) @string.special
```
Comment (theHamsta) [inline queries/t32/highlights.scm:65]: Perhaps`@symbol` applies here. But I don't know the scripting language

**Review by theHamsta — APPROVED** [review]

File: queries/t32/highlights.scm (hunk 1 above)
Comment (xasc) [reply queries/t32/highlights.scm]: Thank you for your feedback. `(symbol)` in this context refers to a debug symbol that is part of a binary for an embedded target. Most of the time debug symbols are acting as a shorthand name for a specific memory address or address range, so symbol probably makes sense here.

On the other hand, paths are very similar to a string without any delimiters. The PRACTICE parse uses the command context to determine whether the next argument is supposed to be a file path and accepts both a string and a path.

**Comment by xasc** [comment]
The latest set of changes should resolve the incorrect queries for locals. I had to introduce new node types, so the change scope got larger than what I had initially hoped for.

**Comment by clason** [comment]
Tests are failing due to wrong capture mismatch with the parser.

**Comment by xasc** [comment]
Thank you very much for the hint. The last grammar update was missing the updated parser sources. Both parts should be in sync now.

**Comment by clason** [comment]
Unfortunately, that didn't do the trick...

**Comment by clason** [comment]
(Also, just a note for future PRs -- not just to this repo: Never make a PR straight from your fork's master, but always open a feature branch and make the PR from there. Otherwise you trigger Github's paranoia and some things will require additional steps.)

**Comment by xasc** [comment]
Looks like as if an older version of the grammar was getting pulled. I guess this was caused by the outdated entry in _lockfile.json_.

**Comment by clason** [comment]
> Looks like as if an older version of the grammar was getting pulled. I guess this was caused by the outdated entry in _lockfile.json_.

Yes, an entry in the `lockfile.json` will mean that this exact commit is used. (It's still good practice to add the new parser with the correct hash to the initial PR to make it self-contained; otherwise we'll have to run a few automated PRs afterwards.)

**Comment by xasc** [comment]
Thank you for the clarification. The grammar is now pinned to the latest compatible version.

**Comment by clason** [comment]
Note that this will _not_ lock the parser for ever; there's an automated CI script that runs (at least) daily and updates the commit hashes in the lockfile to the latest HEAD. If you don't want that, you need to pin the parser (to a tag) in the `parsers.lua` entry.

**Comment by clason** [comment]
@theHamsta any idea why the Windows-CL check keeps failing?

**Comment by xasc** [comment]
Doesn't seem like the MSVC CI job is currently catching build errors:
```
2022-12-16T21:50:59.9826050Z cmd:  pushd C:\Users\runneradmin\AppData\Local\nvim-data\tree-sitter-t32 & gcc  -o parser.so -I./src src/parser.c src/scanner.c -shared -Os -lstdc++ & popd
2022-12-16T21:50:59.9826485Z 
2022-12-16T21:51:01.2720860Z Compiling...
2022-12-16T21:51:01.2721231Z 
2022-12-16T21:51:01.2722399Z ret:  
2022-12-16T21:51:01.2722641Z 
2022-12-16T21:51:01.2722928Z cmd.err:  Error during compilation
2022-12-16T21:51:01.2723186Z 
2022-12-16T21:51:01.2723399Z vim.v.shell_error:  0
2022-12-16T21:51:01.2723637Z 
....
2022-12-16T21:51:01.3629270Z 
2022-12-16T21:51:01.4276964Z Treesitter parser for t32 has been installed
```

The compile stepsis passed without error, but then the test step will fail, because the parser is not available.

**Comment by theHamsta** [comment]
Please don't merge master into your branch! Always rebase.

**Comment by xasc** [comment]
All right. Thank you very much.

**Pull request merged**